test/com/minseungkwon/datastructure/String/CheckStringTest.java

```diff
@@ -48,4 +48,34 @@ public void test_permutationUsingArray() {
         assertTrue(cs.permutationUsingArray("ABC", "BCA"));
         assertFalse(cs.permutationUsingArray("ABC", "BDA"));
     }
+
+    @Test
+    public void test_urlify() {
+        CheckString cs = new CheckString();
+        assertEquals("Mr%20John%20Smith", cs.urlify("Mr John Smith    ", 13));
```

Comment:
Can we make it work without trailing spaces? I think we can make a temp char array from CheckString:93